### DISCUSSION TITLE
Question: How to Use Button in AutoSuggestBox item?

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/microsoft-ui-xaml#7440] [Q&A]
I want the AutoSuggestBox items to have a button, i used a **AppBarButton** Everything works fine except that it does not look good at all
```
<AppBarButton HorizontalAlignment="Right"
                          VerticalAlignment="Center"
                          HorizontalContentAlignment="Stretch"
                          VerticalContentAlignment="Stretch"
                          Click="DeleteHistory_Click"
                          Icon="Delete" />
```
![1](https://user-images.githubusercontent.com/9213496/136651946-faed04a9-6d4a-4472-ae5d-c31c1c14a755.png)

So I used **Button** instead of **AppBarButton**  now The appearance is correct and excellent
![2](https://user-images.githubusercontent.com/9213496/136652032-27f2fb87-1848-4f32-821b-137b0c169226.png)

But a new problem arises and when I click on the button, autosuggestbox Closes and also button click is not working but this problem does not exist in the AppBarButton 
So how can I click on the button without selecting the AutoSuggestBox item?

**Comment by StephenLPeters** [comment]
@ranjeshj @llongley or @kmahone have any of you all examined this before?

**Comment by ghost1372** [comment]
any help?

**Comment by tuggernuts** [comment]
Well, I just ran into this same issue. The item is selected before the click is registered. Neither a click event nor a command is triggered.

In my case, I want a button in the "Nothing found" item, and I'm using an itemtemplateselector for that.

Has anyone found a workaround?

**Comment by gautambjain** [comment]
Experiencing the same issue. Any solution or workaround please?

**Comment by gautambjain** [comment]
I accidentally found the below workaround.

I set the ItemsSource to delete the item from the drop down list. This is not the real workaround.

Setting the text box to blank text surprisingly keeps the drop down list open and also removes the item from the drop down. For now, I will manage with this. But it is little ugly as sends multiple text change events.

```
if (FileSearchTextBox.ItemsSource != null)
{
      FileSearchTextBox.ItemsSource = ((IEnumerable<Tag>)FileSearchTextBox.ItemsSource).Where(t => t != tag);
}
FileSearchTextBox.Text = "";
```

**Comment by deckerth** [comment]
I do not understand why the workaround should help. If the button does not react, how can the user trigger the deletion of an item from the suggestion list so that it never shows up again?
